### PR TITLE
install: inherit stdout/stderr from root lifecycle scripts

### PR DESCRIPTION
**Pull request by dylan-conway**
### What does this PR do?
Changes root lifecycle scripts to print stdout/stderr. Output will print before the install summary.

Screenshot 2024-03-25 at 1 48 03 PM

### How did you verify your code works?
added tests

### CODE REVIEW

File: src/cli/pm_trusted_command.zig [hunk 1]
```diff
@@ -349,7 +349,7 @@ pub const TrustCommand = struct {
                     }
 
                     switch (pm.options.log_level) {
-                        inline else => |log_level| try pm.spawnPackageLifecycleScripts(ctx, info.scripts_list, log_level),
+                        inline else => |log_level| try pm.spawnPackageLifecycleScripts(ctx, info.scripts_list, log_level, false),
```
Comment (jdalton) [inline src/cli/pm_trusted_command.zig]: Can you add a `/* what this param is */ false)` to the above. This hits on the boolean trap: https://ariya.io/2011/08/hall-of-api-shame-boolean-trap

File: src/install/install.zig [hunk 2]
```diff
@@ -9512,18 +9500,21 @@ pub const PackageManager = struct {
...
+                }
             }
 
             while (this.pending_lifecycle_script_tasks.load(.Monotonic) > 0) {
                 if (PackageManager.verbose_install) {
                     if (PackageManager.hasEnoughTimePassedBetweenWaitingMessages()) Output.prettyErrorln("<d>[PackageManager]<r> waiting for {d} scripts\n", .{this.pending_lifecycle_script_tasks.load(.Monotonic)});
                 }
 
-                PackageManager.instance.sleep();
+                this.sleep();
```
Comment (jdalton) [inline src/install/install.zig:9523]: What does sleeping allow?

File: src/install/install.zig [hunk 3]
```diff
@@ -10163,6 +10154,32 @@ pub const PackageManager = struct {
             }
         }
 
+        if (FeatureFlags.breaking_changes_1_1_0) {
```
Comment (jdalton) [inline src/install/install.zig]:
```suggestion
        if (comptime FeatureFlags.breaking_changes_1_1_0) {
```
comtiming ?

File: test/cli/install/registry/bun-install-registry.test.ts [hunk 4]
```diff
@@ -3658,94 +3658,103 @@ for (const forceWaiterThread of [false, true]) {
...
-          name: "foo",
-          version: "1.2.3",
-          dependencies: {
-            "uses-what-bin": "1.0.0",
-            // fake electron package because it's in the default trustedDependencies list
-            "electron": "1.0.0",
-          },
-        }),
-      );
+    test.if(breakingChanges_1_0_0)(
```
Comment (jdalton) [inline test/cli/install/registry/bun-install-registry.test.ts]: Is this breakingChanges_1_0_0 or breakingChanges_1_1_0?

File: src/install/install.zig (hunk 2 above)
Comment (dylan-conway) [reply src/install/install.zig:9523]: While we wait for lifecycle scripts to finish, sleeping prevents us from using 100% of the cpu

File: src/cli/pm_trusted_command.zig (hunk 1 above)
Comment (dylan-conway) [reply src/cli/pm_trusted_command.zig]: Updated. Assigned to variable instead of comment

File: src/install/install.zig (hunk 3 above)
Comment (dylan-conway) [reply src/install/install.zig]: Updated 👍

File: test/cli/install/registry/bun-install-registry.test.ts (hunk 4 above)
Comment (dylan-conway) [reply test/cli/install/registry/bun-install-registry.test.ts]: lol yes this should be 1_1_0

**Comment by dylan-conway** [comment]
Merging into https://github.com/oven-sh/bun/pull/9724

**Pull request closed without merging**